src/components/Editor/EditorMenu.js

```diff
@@ -78,7 +79,10 @@ function getMenuItems(
     label: revealInTreeLabel,
     accesskey: revealInTreeKey,
     disabled: false,
-    click: () => showSource(selectedSource.get("id"))
+    click: () => {
+      clearShownSource();
+      showSource(selectedSource.get("id"));
```

Comment:
If we want to always clear, and then show the source we should move this logic to the showSource action...

In that case we would dispatch two actions one that cleared the source and the other that set it.